.golangci.yml

```diff
@@ -0,0 +1,41 @@
+run:
+  deadline: 12m10s
+  modules-download-mode: readonly
+
+issues:
+  max-per-linter: 0
+  max-same-issues: 0
+
+linters:
+  disable-all: true
+  enable:
+    - asciicheck
+    - deadcode
+    - errcheck
+    - gocritic
+    - gofmt
+    - goimports
+    - gosimple
+    - govet
+    - ineffassign
+    - nakedret
+    - misspell
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - unconvert
+    - varcheck
+    - vet
+    - vetshadow
+    - whitespace
+
+linters-settings:
+  errcheck:
+    ignore: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema:ForceNew|Set,fmt:.*,io:Close
+  misspell:
+    ignore-words:
+      - hdinsight
+      - exportfs
+  gofmt:
+    simplify: false
```

Comment:
This is a general golangci configuration for terraform providers, please advise if any of the checks needs amendment 

Refs.
*  https://github.com/hashicorp/terraform-provider-aws/blob/main/.golangci.yml
*  https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/.golangci.yml
* https://github.com/hashicorp/terraform-provider-google/blob/master/.golangci.yml

Reply:
github action has been removed, please feel free to update zuul config or advise

Reply:
I personally don't like `errcheck` and other rule ignoring.

We currently don't have too many errors with default settings. So maybe just remove the final part? The following:
```yaml
linters-settings:
  errcheck:
    ignore: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema:ForceNew|Set,fmt:.*,io:Close
  misspell:
    ignore-words:
      - hdinsight
      - exportfs
  gofmt:
    simplify: false
```

Then, as part of another issue/PR we will just fix all our errors (I got an only a bit more than 100 with default rules)

Reply:
removed 

```yaml
  misspell:
    ignore-words:
      - hdinsight
      - exportfs
  gofmt:
    simplify: false
```

I kept errorcheck ignores as removing them will produce false alarms in these packages

```yaml
  errcheck:
    ignore: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema:ForceNew|Set,fmt:.*,io:Close
```

Reply:
>I kept errorcheck ignores as removing them will produce false alarms in these packages

Those are not really false alarms :) Not for us, at least. We try to get rid of all unhandled errors.

Reply:
if we enable alrams for schema.Set for example, you would need to change all e.g.

```golang
	d.Set("floating_ip", floatingIP)
	d.Set("instance_id", instanceId)
...
```

to

```
	_ := d.Set("floating_ip", floatingIP)
	_ := d.Set("instance_id", instanceId)
...
```

Reply:
And we do actually handle those returned errors using `multierror` (and rewriting already existing `d.Set` usages). Even considering that `d.Set` returns an error instead of panicking only in the case of acceptance tests.

Reply:
errcheck ignores have been removed.